apps/prairielearn/elements/pl-external-grader-results/pl-external-grader-results.py

```diff
@@ -39,129 +41,140 @@ def ansi_to_html(output):
-                        test["has_message"] = bool(results_test.get("message", None))
-                        test["message"] = ansi_to_html(
-                            results_test.get("message", None)
-                        )
-                        test["has_output"] = bool(results_test.get("output", None))
-                        test["output"] = ansi_to_html(results_test.get("output", None))
-                        test["has_description"] = bool(
-                            results_test.get("description", None)
-                        )
-                        test["description"] = results_test.get("description", None)
-                        test["show_points"] = not tests_missing_points
-                        if not tests_missing_points:
-                            test["points"] = round_value(results_test.get("points"))
-                            test["max_points"] = round_value(
-                                results_test.get("max_points")
-                            )
-
-                            if test["max_points"] != test["points"]:
-                                test["results_color"] = "#E70000"
-                                test["results_icon"] = "fa-times"
-                            # Don't consider points for test cases that are 0/0
-                            # to be correct. We compare with a string because
-                            # `round_value()` returns a string.
-                            elif test["max_points"] == "0":
-                                test["results_color"] = "#757575"
-                                test["results_icon"] = "fa-circle-info"
-                            else:
-                                test["results_color"] = "#008900"
-                                test["results_icon"] = "fa-check"
-
-                        test["images"] = results_test.get("images", [])
-                        test["has_images"] = len(test["images"]) > 0
-                        tests.append(test)
-
-                    html_params["tests"] = tests
-
-        with open("pl-external-grader-results.mustache", "r", encoding="utf-8") as f:
-            html = chevron.render(f, html_params).strip()
-    else:
-        html = ""
-
-    return html
+def round_value(value: float, digits: int = 2) -> str:
+    """
+    Round the given value to the specified precision; default is 2 digits.
+
+    Remove trailing 0s and '.'s, e.g., convert "1.00" to "1".
+    """
+    return f"{value:.{digits}f}".rstrip("0").rstrip(".")
+
+
+def render(element_html: str, data: pl.QuestionData) -> str:
+    # Early-exit if not the submission panel
+    if data["panel"] != "submission":
+        return ""
+
+    feedback = data["feedback"]
+    results = feedback.get("results", None)
+
+    format_errors = data.get("format_errors", {})
+    grader_format_errors = format_errors.get("_external_grader", [])
+
+    grading_succeeded = bool(feedback.get("succeeded", None))
```

Comment:
What type is in `feedback` under the `"succeeded"` key? Seems a little awkward to cast the output of this to a bool instead of checking for a specific value or key.

Reply:
Agreed it's a bit strange, but I'm not keen to change (unless we're sure of some specific value). I can try and take a look for one.

Reply:
Makes sense, if this logic was here before then I'm fine not changing it up too much

Reply:
(It's something like `color(srgb 1 0 0 / 1)` and works in the browser. See [Coloraide's docs](https://facelessuser.github.io/coloraide/color/).)